R/evaluate-promise.R

```diff
@@ -39,9 +39,12 @@ evaluate_promise <- function(code, print = FALSE) {
   temp <- file()
   on.exit(close(temp))
 
-  result <- with_sink(temp,
-    withCallingHandlers(
-      withVisible(code), warning = wHandler, message = mHandler
+  with_reporter(
```

Comment:
What about wrapping a `tryCatch` outside of the `with_sink()` so it can capture and rethrow the error?